### ISSUE TITLE
iOS Build crashes

### ISSUE THREAD

**Issue by midhunevolvier** [dooboolab/react-native-iap#2149]
**Description**

iOS build crashes : Heterogeneous collection literal could only be inferred to '[String : Any]'; add explicit type annotation if this is intentional.

**Expected Behavior**

Build should be success

**Screenshots**
<img width="1420" alt="Screenshot 2022-12-03 at 5 22 12 PM" src="https://user-images.githubusercontent.com/76942076/205439627-70587011-f6e4-4584-ae25-619c669af8a8.png">

**Environment:**

- react-native-iap: "^12.4.8"
- react-native: "0.70.5"
- Platforms (iOS, Android, emulator, simulator, device): iOS

**To Reproduce**

Just build the app

**Comment by fitimbyttyqi** [comment]
@midhunevolvier I can confirm this error is happening to me as well with the above version of the package.

**Comment by fitimbyttyqi** [comment]
@midhunevolvier I can confirm the issue is happening to me too.

Seems like the issue is at line 799 and 807 from the above screenshot you have provided, 

Not an ios developer but I think changing:
```
"transactionDate": transaction.transactionDate?.millisecondsSince1970 
```
with
```
"transactionDate": transaction.transactionDate?.millisecondsSince1970String 
```

Until any of the package maintainers fixes this you could use version 12.0.0 which I guess works correctly.

**Comment by midhunevolvier** [comment]
Thank you @fitimbyttyqi for the solution, build success.

**Comment by andresesfm** [comment]
@fitimbyttyqi @midhunevolvier . Thank you for the report. I don't see that error locally. Nor it fails in our CI build. Can you please tell us if you are overriding something locally such as the swift version you are using?

**Comment by andresesfm** [comment]
Released on: `12.4.9`
